pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1160,7 +1113,60 @@ public void addTable(@Nonnull TableConfig tableConfig)
         throw new InvalidTableConfigException("UnSupported table type: " + tableType);
     }
 
-    handleBrokerResource(tableNameWithType, brokersForTenant);
+    String brokerTenantName = TagNameUtils.getBrokerTagForTenant(tableConfig.getTenantConfig().getBroker());
+    handleBrokerResource(tableNameWithType, HelixHelper.getInstancesWithTag(_helixZkManager, brokerTenantName));
+  }
+
+  /**
+   * Validates the tenant config for the table
+   */
+  private void validateTableTenantConfig(TableConfig tableConfig, String tableNameWithType, TableType tableType) {
```

Comment:
Will be great if you can add a unit test for this. It will mean mocking the helix get tagged instances call, I suppose.

Reply:
This whole method is not something I've added. I've just wrapped the code into a method. And I can see addTable is used in multiple tests. I'll check what all those cover and add if specific cases are missing